File: 0082-remove-duplicates-from-sorted-list-ii/0082-remove-duplicates-from-sorted-list-ii.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        HashSet<Integer> hashset = new HashSet<>();
        HashSet<Integer> hashsettemp = new HashSet<>();
        ListNode dummy = new ListNode();
        ListNode start = dummy;
        ListNode temp = head;
        
        while(temp != null) {
            if(hashset.contains(temp.val)) {
                hashsettemp.add(temp.val);
            }
            hashset.add(temp.val);
            temp = temp.next;
        }
        for(int i : hashsettemp) {
            hashset.remove(i);
        }
        
        List<Integer> list = new ArrayList<>(hashset);
        Collections.sort(list);
        for(int i : list) {
            start.next = new ListNode(i);
            start = start.next;
        }
        return dummy.next;
    }
}

File: 0082-remove-duplicates-from-sorted-list-ii/0082-remove-duplicates-from-sorted-list-ii.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        hashset = set()
        hashsettemp = set()
        cur = head
        dummy = ListNode()
        temp = dummy
        res = []
        
        while cur:
            if cur.val in hashset:
                hashsettemp.add(cur.val)
            else:
                hashset.add(cur.val)
            cur = cur.next
            
        for i in hashset:
            if i not in hashsettemp:
                res.append(i)
                
        res = sorted(res)
        for i in res:
            temp.next = ListNode(i)
            temp = temp.next
        return dummy.next